app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabsListItem.kt

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray.syncedtabs
+
+import androidx.navigation.NavController
+import mozilla.components.feature.syncedtabs.view.SyncedTabsView
+
+/**
+ * The various types of list items that can be found in a [SyncedTabsList].
+ */
+sealed class SyncedTabsListItem {
+
+    /**
+     * A device header for displaying a synced device.
+     */
+    data class Device(val displayName: String, val id: String) : SyncedTabsListItem()
+
+    /**
+     * A tab that was synced.
+     */
+    data class Tab(val tab: mozilla.components.browser.storage.sync.Tab) : SyncedTabsListItem()
```

Comment:
nit: import aliases can make this easier to read.

Example..

```suggestion
import mozilla.components.browser.storage.sync.Tab as SyncTab
    data class Tab(val tab: SyncTab) : SyncedTabsListItem()
```